### ISSUE TITLE
[Bug] Can't build openvino 2022.1 with GPU in Ubuntu 22.04 and clang14

### ISSUE THREAD

**Issue by sorny92** [openvinotoolkit/openvino#12692]
##### System information (version)
- OpenVINO => 2022.1
- Operating System / Platform => Ubuntu 22.04
- Compiler => Clang 14
- Problem classification: Build error
##### Steps to reproduce

Trying to build openvino in Ubuntu 22.04 with Clang-14.
I try to build it with CPU and GPU backends without VPU or GNA modules.

If I disable `ENABLE_INTEL_GPU` I can get it to build but without GPU support.

At some point I get this error:
```
/home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_debug.hpp:48:6: error: 'DNNL_ENABLE_MEM_DEBUG' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (DNNL_ENABLE_MEM_DEBUG == DNNL_MEM_DEBUG_UNDERFLOW)
     ^
[  7%] Building CXX object src/plugins/intel_gpu/src/kernel_selector/CMakeFiles/openvino_intel_gpu_kernels.dir/common/tensor_type.cpp.o
[  7%] Building CXX object src/plugins/intel_cpu/thirdparty/mkl-dnn/src/common/CMakeFiles/dnnl_common.dir/binary.cpp.o
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/gpu/gpu_batch_normalization_list.cpp:17:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/gpu/gpu_impl_list.hpp:23:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/engine.hpp:33:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/primitive_desc.hpp:27:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_tracking.hpp:23:
/home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_debug.hpp:48:6: error: 'DNNL_ENABLE_MEM_DEBUG' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (DNNL_ENABLE_MEM_DEBUG == DNNL_MEM_DEBUG_UNDERFLOW)
     ^
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/cache_blob_id.cpp:20:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/engine.hpp:33:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/primitive_desc.hpp:27:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_tracking.hpp:23:
/home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_debug.hpp:48:6: error: 'DNNL_ENABLE_MEM_DEBUG' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (DNNL_ENABLE_MEM_DEBUG == DNNL_MEM_DEBUG_UNDERFLOW)
     ^
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/gpu/compute/compute_engine.cpp:20:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/gpu/compute/compute_engine.hpp:26:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/engine.hpp:33:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/primitive_desc.hpp:27:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_tracking.hpp:23:
/home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_debug.hpp:48:6: error: 'DNNL_ENABLE_MEM_DEBUG' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (DNNL_ENABLE_MEM_DEBUG == DNNL_MEM_DEBUG_UNDERFLOW)
     ^
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/gpu/ocl/capi/engine.cpp:22:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/engine.hpp:33:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/primitive_desc.hpp:27:
In file included from /home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_tracking.hpp:23:
/home/esteve/libraries/openvino/src/plugins/intel_gpu/thirdparty/onednn_gpu/src/common/memory_debug.hpp:48:6: error: 'DNNL_ENABLE_MEM_DEBUG' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (DNNL_ENABLE_MEM_DEBUG == DNNL_MEM_DEBUG_UNDERFLOW)
     ^
1 error generated.

```

**Comment by jgespino** [comment]
@sorny92 Building OpenVINO from source has not been validated on Ubuntu 22.04 as mentioned in the [build guide](https://github.com/openvinotoolkit/openvino/wiki/BuildingForLinux#build-on-linux-systems). Can you try building on one of the validated OS and see you run into the same issue?

**Comment by ilya-lavrenov** [comment]
Fixed in current master (2023.0 release)
